source/base/subscriptor.cc

```diff
@@ -175,7 +158,6 @@ Subscriptor::subscribe(const char *id) const
 #ifdef DEBUG
   if (object_info == 0)
     object_info = &typeid(*this);
-  Threads::Mutex::ScopedLock lock (subscription_lock);
   ++counter;
```

Comment:
The criticism wasn't directed at you, but at the original author of that comment.